hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileReaderImpl.java

```diff
@@ -1341,8 +1342,9 @@ public HFileBlock readBlock(long dataBlockOffset, long onDiskBlockSize, final bo
         // Cache the block if necessary
         cacheConf.getBlockCache().ifPresent(cache -> {
           if (cacheBlock && cacheConf.shouldCacheBlockOnRead(category)) {
+            // Using the wait on cache during compaction and prefetching.
             cache.cacheBlock(cacheKey, cacheCompressed ? hfileBlock : unpacked,
-              cacheConf.isInMemory());
+              cacheConf.isInMemory(), isCompaction || cacheOnly);
```

Comment:
Sorry to come back/forth here, but I guess it's better not to cache blocks we are reading during compaction, as these may soon require eviction once compaction is finished and it may just waste space in the cache we could use for the blocks we are writing during compaction.

Reply:
Removed the compaction flog from the check. Yeah it's better not to cache blocks reading during compaction or evict soon.